### PR TITLE
Add CI tests with minimum supported versions

### PR DESCRIPTION
**Pull request by jrbourbeau**
xref https://github.com/nsidc/earthaccess/issues/386#issuecomment-1841302469

Closes https://github.com/nsidc/earthaccess/issues/386

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,10 @@
+[project]
+name = "earthaccess"
+version = "0.8.1"
+
+[tool.setuptools]
+packages = ["earthaccess"]
+
```
Comment (jrbourbeau) [inline pyproject.toml:7]: The changes in this file were needed to make `pip install -e .` work. Open to suggestions from others on a better way to do this (this was me just working through errors until things worked)

**Comment by jrbourbeau** [comment]
Note we're now getting 

```
FAILED tests/unit/test_results.py::test_data_links - AttributeError: 'DataGranules' object has no attribute 'headers'
```

Like reported in https://github.com/nsidc/earthaccess/issues/386

**Review by MattF-NSIDC — APPROVED** [review]
I think this looks awesome. I'd really like some documentation about how this file should be maintained before merging it in. A comment block at the top, or a mention in `CONTRIBUTING.md`, or a PR template with a checklist, or ...?

File: pyproject.toml (hunk 1 above)
Comment (jhkennedy) [reply pyproject.toml:7]: I don't think we need setuptools *at all* and could/should just remove `setup.py`. Notably, it states:
```
# This is a shim to hopefully allow Github to detect the package, build is done with poetry
```
https://github.com/nsidc/earthaccess/blob/main/setup.py

It was added in #181 by @betolink  -- Luis was this just so GitHub can recognise [who depends on it](https://github.com/nsidc/earthaccess/network/dependents)?

**Comment by jhkennedy** [comment]
> Thoughts about keeping the mindeps file up-to-date?

It we were using setuptools, we could do something like this:
* `pyproject.toml`
  ```toml
  [tool.setuptools.dynamic]
  dependencies = { file = ["ci/requirements-min.txt"] }
  ```
* test env using conda or pip:
  ```
  conda create --name earthaccess --file  ci/requirements-min.txt
  ```
  ```
  python -m pip install ci/requirements-min.txt
  ```
I'm not sure if you can do the same with Poetry

File: pyproject.toml (hunk 1 above)
Comment (jhkennedy) [reply pyproject.toml:7]: Okay, if I checkout this branch, remove these changes to `pyproject.toml` (other than the deps update), and remove `setup.py`, this is the error message from pip:
```
ERROR: Project file://.../earthdata has a 'pyproject.toml' and its build backend is missing the 'build_editable' hook. Since it does not have a 'setup.py' nor a 'setup.cfg', it cannot be installed in editable mode. Consider using a build backend that supports PEP 660.
```

If I change the build system to:
```yaml
[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```
as recommended here: https://python-poetry.org/docs/pyproject#poetry-and-pep-517

it works

Comment (jhkennedy) [reply pyproject.toml]: Looks good!

**Comment by jhkennedy** [comment]
> For us I think just manually updating is fine.

I agree with this as well. I like the comments in both files, and I don't think it's a burden since we should rarely be updating those.

**Comment by jrbourbeau** [comment]
Thanks all for the reviews!
